apps/eth-wrapper/src/components/EthBalance.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useState } from 'react';
+import { ethers } from 'ethers';
+import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
+import { Text } from '@gnosis.pm/safe-react-components';
+
+const EthBalance: React.FC = () => {
+    const [balance, setBalance] = useState("loading...");
+    const { sdk, safe } = useSafeAppsSDK();
+
+    async function fetchBalance() {
+        const balanceEth = await sdk.eth.getBalance([safe.safeAddress]);
+        setBalance(ethers.utils.formatEther(balanceEth));
+    }
+
+    useEffect(() => {
+        fetchBalance();
+        console.log("Updating ETH balance");
+    });
```

Comment:
shouldn't this have a dependency array?

Reply:
This is just the label with the balance. You would have to navigate away from the `eth-wrapper` app , to run a transaction and affect your balance. Unless, it is expected that I keep this value updated, in which case the dependency would be `sdk`?